examples/react/src/app/services/posts.ts

```diff
@@ -14,7 +14,10 @@ export const postApi = createApi({
   endpoints: (build) => ({
     getPosts: build.query<PostsResponse, void>({
       query: () => 'posts',
-      provides: (result) => result.map(({ id }) => ({ type: 'Posts', id })),
+      provides: (result) => [
+        ...result.map(({ id }) => ({ type: 'Posts', id } as const)),
+        { type: 'Posts', id: 'LIST' },

```

Comment:
:question: why is this "LIST" bit needed?

Reply:
If you invalidate `['Posts']`, it will cause _every query_ that provides a type of 'Post' to be rerun, even if it provides a specific `id`. So in the example of this demo, the way it was set up originally was that when you added a new post, it would cause `PostDetail` to switch into a loading state, which is not what we want. So switching to 'LIST' here causes the `addPost` mutation to _only_ rerun `getPosts`, and `getPost(2)` remains stable as it's data doesn't change.

Does that make sense? :D

Reply:
so is "LIST" a magic thing for queries in general, or something specific to this particular demo?  Is this behavior documented somewhere?

Reply:
It's just an id, like 5 or 7 would be, with a descriptive name. Nothing forces you to have those `{ type: 'Posts', id: something }` correlate to anything.

We have the special situation here that adding a new element should not refetch any existing id, but that it should refetch all lists (because it might be somewhere on there). So all lists `provide` the ids they really have, plus LIST - and the add endpoint invalidates that.

Of course, you could also introduce a new type PostsList for that or something.

Reply:
'LIST' is just a generic id that applies to any query --  you could also call it '*', 'ALL' or 'BANANA' if you wanted. In theory, you could create multiple ids like 'SIDEBAR_LIST', 'PAGE_LIST', and if for some reason you wanted to... you could only refresh one of those two when adding a post.

It's hinted at in the docs in the [example here](https://rtk-query-docs.netlify.app/concepts/mutations#advanced-mutations-with-revalidation), but I need to add more copy spelling out the exact behavior and demonstrate further use cases.

Reply:
And yeah, these little tricks should probably make it into the advanced invalidation section. Right now, it's only mentioned in the example in the readme I believe.

Reply:
Right, that's kinda my point - I wasn't sure if it was something unique to this demo, or built into the library itself, or a fallout of something that's inherent in the lib.

Reply:
Example scenarios

**Invalidating everything**
**queries:**
getPosts: provides ['Posts']
getPost: provides [{ type: 'Posts', id }]
**mutations:**
addPost: invalidates ['Posts']

**Component:**
<App>
<AddPost />
<PostsList />
<PostDetail id={id} />
<PostDetail id={id} />
<PostDetail id={id} />
</App>

Behavior: 
When you `addPost`, it will cause each `PostDetail` component to go into a loading state because `addPost` invalidates the root entity, which causes any query that provides 'Posts' to be re-run.


**Selectively invalidating lists**
**queries:**
getPosts: provides [{ type: 'Posts', id: 'LIST' ]
getPost: provides [{ type: 'Posts', id }]
**mutations:**
addPost: invalidates [{'Posts', id: 'LIST'}]

**Component:**
<App>
<AddPost />
<PostsList />
<PostDetail id={id} />
<PostDetail id={id} />
<PostDetail id={id} />
</App>

Behavior: 
When you `addPost`, it will only cause the `PostsList` to into a loading state because `addPost` invalidates the 'LIST' id, which causes `getPosts` to rerun (because it provides that specific id).

Reply:
And yea, docs are still a WIP, but I'll get them added in. There is also the case that this may end up not being a concern for folks if we do background data fetching and are able to differentiate between `loading` and `background fetching`. Either way, I do need to spell out the network request behavior :)

Reply:
Okay! Now we have 'background fetching' 🎉